bundle-workflow/src/test_workflow/local_test_cluster.py

```diff
@@ -0,0 +1,105 @@
+        if not self.security_enabled:
+            self.disable_security(dir)
+        self.process = subprocess.Popen(
+            "./opensearch-tar-install.sh",
+            cwd=dir,
+            shell=True,
+            stdout=self.stdout,
+            stderr=self.stderr,
+        )
+        print(f"Started OpenSearch with PID {self.process.pid}")
+        self.wait_for_service()
+
+    def endpoint(self):
+        return "localhost"
+
+    def port(self):
+        return 9200
+
+    def destroy(self):
+        if self.process is None:
+            print("Local test cluster is not started")
+            return
+        print(f"Sending SIGTERM to PID {self.process.pid}")
+        self.process.terminate()
+        try:
+            print("Waiting for process to terminate")
+            self.process.wait(10)
+        except subprocess.TimeoutExpired:
+            print("Process did not terminate after 10 seconds. Sending SIGKILL")
+            self.process.kill()
+            try:
+                print("Waiting for process to terminate")
+                self.process.wait(10)
+            except subprocess.TimeoutExpired:
+                print("Process failed to terminate even after SIGKILL")
+                raise
+        finally:
+            print(f"Process terminated with exit code {self.process.returncode}")
+            self.stdout.close()
+            self.stderr.close()
+            self.process = None
+
+    def url(self, path=""):
+        return f'{"https" if self.security_enabled else "http"}://{self.endpoint()}:{self.port()}{path}'
+
+    def download(self):
+        print(f"Creating local test cluster in {self.work_dir}")
+        os.chdir(self.work_dir)
+        print(f"Downloading bundle from {self.manifest.build.location}")
+        urllib.request.urlretrieve(self.manifest.build.location, "bundle.tgz")
+        print(f'Downloaded bundle to {os.path.realpath("bundle.tgz")}')
+
+        print("Unpacking")
+        subprocess.check_call("tar -xzf bundle.tgz", shell=True)
+        print("Unpacked")
+
+    def disable_security(self, dir):
+        subprocess.check_call(
+            f'echo "plugins.security.disabled: true" >> {os.path.join(dir, "config", "opensearch.yml")}',
+            shell=True,
+        )
+
+    def wait_for_service(self):
+        print("Waiting for service to become available")
+        url = self.url("/_cluster/health")
+
+        for attempt in range(10):
+            try:
+                print(f"Pinging {url} attempt {attempt}")
+                response = requests.get(url, verify=False, auth=("admin", "admin"))
+                print(f"{response.status_code}: {response.text}")
+                if response.status_code == 200 and '"status":"green"' in response.text:
+                    print("Cluster is green")
+                    return
+            except requests.exceptions.ConnectionError:
+                print("Service not available yet")
+            time.sleep(10)
```

Comment:
Is there any reason why we attempt to check at equal intervals instead of using an exponential backoff strategy for retrying requests?

Reply:
In this PR this file only moved from `python` to `src` and hasn't changed. 

Reading the code, there's no real good reason for timeout choices nor does it matter much. We're booting a service, then waiting for it to start. If anything, I would exponentially forward-on here because after waiting a long time it's likely the service will start "sooner".